azavea/pfb-network-connectivity
# Facilitate UI-driven analyses for non-US places (#889) (#890) (#891)

## Overview

The client needs to be able to run analyses using non-US places. To run a successful analysis for a non-US place, a custom population file and jobs file will need to be provided to the analysis script. An OSM extract may be needed as well if the file cannot be found at the geofabrik URL, which we derive the country's alpha3 code. This PR parameterizes the analysis script to fulfill those needs, then extends those parameters to the UI through the "Run Analysis" button in the Analysis Jobs admin page.

### Demo

![Screenshot from 2022-07-05 13-46-20](https://user-images.githubusercontent.com/36001800/177387998-c7d0ecc0-4cd0-4fb2-86f0-32695b1bfc92.png)


### Notes

We had to increase the allowable geographic width of a shapefile in detect_utm_zone.


## Testing Instructions

* make sure you have a few hours in which you can let this run
* run `scripts/setup`, `vagrant ssh`, `scripts/server`
* navigate to http://localhost:9301/admin#/admin/neighborhoods/create/
* create a neighborhood for Brussels using this [zip](https://drive.google.com/file/d/1Ge9dvSS9kjWKg5C5sNt18fZuN1NYcxBI/view?usp=sharing)
* navigate to http://localhost:9301/admin#/admin/analysis-jobs/create/
* create an analysis using
  * a link to [this population file](https://drive.google.com/file/d/1rwQtUgfTyBKpBKGhVpKI8_fYLVFA-B-X/view?usp=sharing)
    * this will require downloading the file, uploading it to your s3 bucket, then dropping that link in
    * or you can just use [mine](https://jacob-pfb-storage-us-east-1.s3.amazonaws.com/data/tabblock2010_98_pophu.zip)
  * and a link to [this jobs file](https://drive.google.com/file/d/1EkNW-zrViUGTjIvVHMcTJQDu3aB2hlS1/view?usp=sharing)
    * same as above
    * or you can just use [mine](https://jacob-pfb-storage-us-east-1.s3.amazonaws.com/data/be_od_main_JT00_2017.csv.gz)
  * you don't need to set the OSM extract url for belgium because the analysis can find it on geofabrik but you can!
    * use the [osm extract url from geofabrik](https://download.geofabrik.de/europe/belgium-latest.osm.pbf)
* nearly half a work day later, you should see `Updating job status: COMPLETE Finished analysis` and some scores
* for good measure, navigate to http://localhost:9301/admin#/places//// and click on your neighborhood, then verify that the scores came through (i.e. it's not all 0s)


## Checklist

- [x] Tested without OSM file set
- [x] Tested with OSM file set
- [x] Tested with jobs skipped

Resolves #614, #889, #890, #891
